### PR TITLE
doc/dev: use prompt directive when appropriate

### PR DESCRIPTION
**Pull request by tchaikov**
for 2 reasons:

* sphinx renders codeblock using python syntax highlighting by default,
  so it's not surprising that it highlight keywords like "export" in
  command line samples. so to render command line code blocks, we'd
  better specify the syntax explicitly for better rendering result.
* with the help of "prompt" directive, user is able to copy and paste
  the command without the prompt. for instance, with the default
  "::" directive, user will copy "$ ceph df", which is not very
  convenient, but with "prompt" directive, user only copies
  "ceph df".

Signed-off-by: Kefu Chai <kchai@redhat.com>

## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
jenkins render docs

File: doc/dev/developer_guide/basic-workflow.rst [hunk 1]
```diff
@@ -76,7 +76,7 @@ detailed instructions on forking. In short, if your GitHub username is
 https://github.com/mygithubaccount/ceph. Once you have created your fork,
 you clone it by doing:
 
-.. code::
+.. prompt:: bash $
 
     $ git clone https://github.com/mygithubaccount/ceph
```
Comment (jdurgin) [inline doc/dev/developer_guide/basic-workflow.rst:81]: no longer need the $ here and a few other places below - it shows up doubled

Comment (tchaikov) [reply doc/dev/developer_guide/basic-workflow.rst]: @jdurgin fixed and repushed.

**Comment by tchaikov** [comment]
jenkins render docs

**Comment by tchaikov** [comment]
jenkins render docs

File: doc/dev/developer_guide/issue-tracker.rst [hunk 2]
```diff
@@ -1,3 +1,5 @@
+.. _issue-tracker:
```
Comment (jdurgin) [inline doc/dev/developer_guide/issue-tracker.rst:1]: looks like this ref didn't work for some reason (is there a dash-space conversion?):

```
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/developer_guide/basic-workflow.rst:198: WARNING: Unknown target name: "issue tracker".
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/developer_guide/basic-workflow.rst:303: WARNING: Unknown target name: "issue tracker".
/home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/internals.rst:18: WARNING: duplicate label mailing-list, other instance in /home/jenkins-build/build/workspace/ceph-pr-docs/doc/dev/developer_guide/essentials.rst```

Comment (tchaikov) [reply doc/dev/developer_guide/issue-tracker.rst:1]: no, my bad. fixed and repushed.

**Review by jdurgin — APPROVED** [review]

**Comment by tchaikov** [comment]
jenkins render docs

**Pull request merged**